src/pages/Signup/Signup.js

```diff
@@ -0,0 +1,208 @@
+import React, { useState } from 'react';
+import './Signup.scss';
+import Input from './components/Input/Input';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import { ko } from 'date-fns/esm/locale';
+
+const Signup = () => {
+  const [userData, setUserData] = useState({ gender: '남성', userType: '1' });
+  const [date, setDate] = useState(new Date());
+  const [checkNickName, setCheckNickName] = useState(false);
+  const handleInput = e => {
+    const { name, value } = e.target;
+
+    if (name === 'phoneNumber') {
+      const value = e.target.value
+        .replace(/[^0-9]/g, '')
+        .replace(/([0-9]{3})([0-9]{3,4})([0-9]{4})/g, '$1-$2-$3');
+      e.target.value = value;
+      setUserData(pre => {
+        return { ...pre, [name]: value };
+      });
+    } else {
+      setUserData(pre => {
+        return { ...pre, [name]: value };
+      });
+    }
+  };
+
+  const handleDate = data => {
+    setDate(data);
+    data = new Date(data);
+    const year = data.getFullYear();
+    const month = data.getMonth() + 1;
+    const date = data.getDate();
+    data = year + '-' + month + '-' + date;
+    setUserData(pre => {
+      return { ...pre, birthday: data };
+    });
+  };
+
+  const handleUserType = e => {
+    setUserData(pre => {
+      return { ...pre, userType: e.target.value };
+    });
+  };
+
+  const handleCheckNickName = () => {
+    setCheckNickName(true);
+  };
+
+  const goSignUp = () => {
+    console.log(userData);
+  };
+
+  const checkAllWrite =
+    userData.userType === '1'
+      ? userData.height &&
+        userData.nickname &&
+        userData.phoneNumber &&
+        userData.weight &&
+        checkNickName &&
+        userData.interestedWorkout &&
+        userData.workoutLoad
+      : userData.height &&
+        userData.nickname &&
+        userData.phoneNumber &&
+        userData.weight &&
+        checkNickName &&
+        userData.interestedWorkout &&
+        userData.workoutLoad &&
+        userData.specialized;
```

Comment:
다른 부분까지 건드리면 좀 더 개선의 여지가 있는 코드지만.. 일단 구현된 대로 여기만 깔끔하게 수정해볼 수 있을 것 같습니다.
```suggestion
  const checkAllWrite =
    userData.height &&
    userData.nickname &&
    userData.phoneNumber &&
    userData.weight &&
    checkNickName &&
    userData.interestedWorkout &&
    userData.workoutLoad &&
    (userData.userType === '1' || userData.specialized);
```